chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -201,7 +201,7 @@ def self.lookup_interface_info(node, conduit, intf_to_if_map = nil)
         return [nil, nil] if intf_to_if_map[conduit].nil?
 
         c_info = intf_to_if_map[conduit]
-        interface_list = c_info["if_list"]
+        interface_list = c_info["if_list"] || []
```

Comment:
I am pretty sure this is not going to fix the problem. If interface_list was nil here before chef would have thrown and "NoMethodError: undefined method `size' for nil:NilClass" exception on line 204. If you initialize to [] it will bail out on  line 219. Unless I am overlooking something.